scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -945,7 +945,8 @@ class FormatOps(
       }
 
   def binPackParentConstructorSplits(
-      chain: Either[Template, Seq[Type.With]],
+      chain: Either[Tree, Seq[Type.With]],
+      inits: Option[List[Init]],
```

Comment:
should we add
```
  require(chain.isRight == inits.isEmpty)
```

Reply:
Maybe we could have an additional method :

```scala
  def binPackParentConstructorSplits(
      chain: Tree
      inits: Option[List[Init]],
      lastToken: Token,
      indentLen: Int) = {
  val extendsThenWith = inits.exists(_.length > 1)
  binPackParentConstructorSplits(Left(chain), extendsThenWith, lastToken, indentLen)
 }
```

and change the second chain parameter to `extendsThenWith : Boolean = false` (we might need to move it last to make sense).
I think it would work a bit nicer than `require`.

Reply:
It would require to overload the `binPackParentConstructorSplits` three times. Moreover we have `splitWithChain` that uses it and thereby should be overloaded too in a similar manner. I would go with @kitbellew suggestion.

Reply:
One overload would be enough I think. `require` will throw a runtime exception, which is why I am not keen on it.